assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,176 @@
+import {
+  addDays,
+  addMonths,
+  addYears,
+  format,
+  parseISO,
+  setHours,
+  setMinutes,
+} from 'date-fns';
+import React, { useState } from 'react';
+import { noop, truncate } from 'lodash';
+import { EOS_CONTENT_COPY, EOS_INFO_OUTLINED } from 'eos-icons-react';
+import Button from '@common/Button';
+import Modal from '@common/Modal';
+import Input from '@common/Input';
+import Label from '@common/Label';
+import Select from '@common/Select';
+import Switch from '@common/Switch';
+
+const normalizeExpiration = (expiration) =>
+  setMinutes(setHours(expiration, 0), 0);
+
+const availableTimeOptions = [
+  {
+    type: 'months',
+    timeGenerator: (quantity) => addMonths(new Date(), quantity),
+  },
+  {
+    type: 'days',
+    timeGenerator: (quantity) => addDays(new Date(), quantity),
+  },
+  {
+    type: 'years',
+    timeGenerator: (quantity) => addYears(new Date(), quantity),
+  },
+];
+
+function ApiKeySettingsModal({
+  open = false,
+  loading = false,
+  onSave = noop,
+  onClose = noop,
+  generatedApiKey,
+  generatedApiKeyExpiration,
+}) {
+  const timeOptions = availableTimeOptions.map((o) => o.type);
+  const [timeQuantity, setTimeQuantity] = useState(0);
+  const [timeQuantityType, setTimeQuantityType] = useState(timeOptions[0]);
+
+  const [quantityError, setQuantityError] = useState(false);
+  const [timeFormEnabled, setTimeFormEnabled] = useState(true);
+
+  const saveFormState = () => {
+    if (!timeFormEnabled) {
+      onSave({ apiKeyExpiration: null });
+      return;
+    }
+    if (timeQuantity === 0 || !timeQuantity) {
+      setQuantityError(true);
+      return;
+    }
+    const timeQuantitySettings = availableTimeOptions.find(
+      (q) => q.type === timeQuantityType
+    );
+    const apiKeyExpiration = timeQuantitySettings.timeGenerator(timeQuantity);
+
+    onSave({ apiKeyExpiration: normalizeExpiration(apiKeyExpiration) });
+  };
+
+  return (
+    <Modal
+      title="API Key Settings"
+      className="!w-3/4 !max-w-3xl"
+      open={open}
+      onClose={onClose}
+    >
+      <div className="flex flex-col my-5">
```

Comment:
Checking the modal in `storybook`, I would say that some margins don't follow the mock up.
Can you confirm this with @jagabomb ?
I'm fine, but I want Jurgen's approval there hehe

Reply:
![API Key Default State](https://github.com/trento-project/web/assets/40714533/82ef3291-aa1d-4e37-91b4-aa4bec934e54)
Hi @CDimonaco, here is my feedback regarding the spacing, and number input for the default state:

1. Reduce and standardise the spacing between the heading and between line of content (highlighted in red block with pixel height dimensions). Ideally they were 16px in the design.
2. Remove the margin below the close button. There is already a spacing which is provided by the padding of the modal.
3. Resolve this UI addition and subtraction indicator for the numeric field since there is a native component (Chrome) which creates a duplication of the same function.

Reply:
For the point 1 and 2, I will standardize the space, for the last point, this is a custom component from the same component library we use for other inputs, if we fallback to the native component, we should design all the style and create a custom component to match all the others input design.

Why this component is bad? We could of course fallback to the native component but we need to create a custom component which I think it's a big effort and out the scope of this pr.

I tested on multiple browser and I cannot reproduce the "dual" indicators on the number component, but again, if we need to write custom logic on top of the custom or native component it's not an easy task and out of the scope of this pr.

cc @dottorblaster maybe my difficulty estimation is wrong, wdyt?

Reply:
@CDimonaco I agree, if the effort is too large for the numeric input to be improved then we can re-look at this in another PR. I just wanted to note the input behaviour since Alberto also raised it too.

Reply:
Point 1 and 2 are perfectly addressable in this PR, when it comes to point 3 I didn't quite get your feedback @jagabomb, my advice is to merge and proceed incrementally to do this kind of work on the design system, which we couldn't foresee.

The alternative is to adopt a text input for the moment, but I would suggest otherwise since with a number input we can ensure nothing strange is submitted.

Reply:
Space standardized, you can check in storybook but this is the screenshot, there are also no more margin under the button, only the modal margins.

![image](https://github.com/trento-project/web/assets/9409502/1e82e2f6-b9fb-4146-bc74-e4bcdb6e3c56)

Reply:
@dottorblaster It is easier to see the issue with the duplicated numeric addition and subtraction (arrow up and arrow down) buttons below. But as I said before, I am happy to defer this to another PR and revisit it when we have time. 😊 
<img width="832" alt="Screenshot 2024-03-13 at 09 34 54" src="https://github.com/trento-project/web/assets/40714533/0bbc7c77-c0de-4c74-868f-be6ab3d706df">